lib/ASTGen/Sources/ASTGen/ParameterClause.swift

```diff
@@ -61,40 +61,63 @@ extension EnumCaseParameterSyntax: ValueParameterSyntax {
 }
 
 extension ASTGenVisitor {
-  func generate(functionParameter node: FunctionParameterSyntax) -> BridgedParamDecl {
-    self.makeParamDecl(node)
+  func generate(functionParameter node: FunctionParameterSyntax, forSubscript: Bool) -> BridgedParamDecl {
+    // For non-subscripts, the argument name is defaulted to the parameter name.
+    self.makeParamDecl(node, argNameByDefault: !forSubscript)
   }
 
   func generate(enumCaseParameter node: EnumCaseParameterSyntax) -> BridgedParamDecl {
-    self.makeParamDecl(node)
+    self.makeParamDecl(node, argNameByDefault: true)
   }
 
-  private func makeParamDecl(_ node: some ValueParameterSyntax) -> BridgedParamDecl {
+  /// Generate a ParamDecl. If `argNameByDefault` is true, then the parameter's
+  /// argument label is inferred from the first name if no second name is present.
+  private func makeParamDecl(_ node: some ValueParameterSyntax, argNameByDefault: Bool) -> BridgedParamDecl {
+    precondition(
+      !(node.optionalFirstName == nil && node.secondName != nil),
+      "If only one name is present, it should be the first"
+    )
```

Comment:
Is this something we can assert, or should we be defensive here? I believe this should only apply to EnumCaseParameterSyntax